pmoves/services/content-provenance-gate/main.py

```
@@ -0,0 +1,741 @@
+
+async def _nats_resilience_loop() -> None:
+    global _nc
+    backoff = 1.0
+
+    while True:
+        nc = NATS()
+        disconnect_event = asyncio.Event()
+
+        def _mark_connection_lost(reason: str) -> None:
+            global _nc
+            if _nc is nc:
```

Ruff (0.15.12)

[warning] Function definition does not bind loop variable `nc`

(B023)